sct-commons/src/test/java/org/lfenergy/compas/sct/commons/scl/ied/LNAdapterTest.java

```diff
@@ -152,33 +157,76 @@ void checkExtRefInfoCoherence() throws Exception {
         extRefInfo.getSignalInfo().setPDO("Do11.sdo11");
         extRefInfo.getSignalInfo().setIntAddr("INT_ADDR11");
 
-        TExtRef extRef = assertDoesNotThrow(() ->lnAdapter.checkExtRefInfoCoherence(extRefInfo));
-        assertEquals(extRefInfo.getSignalInfo().getPDO(),extRef.getPDO());
+        TExtRef extRef = assertDoesNotThrow(() -> lnAdapter.checkExtRefInfoCoherence(extRefInfo));
+        assertEquals(extRefInfo.getSignalInfo().getPDO(), extRef.getPDO());
 
         ExtRefBindingInfo bindingInfo = DTO.createExtRefBindingInfo();
         extRefInfo.setBindingInfo(bindingInfo);
-        assertThrows(ScdException.class, () ->lnAdapter.checkExtRefInfoCoherence(extRefInfo));
+        assertThrows(ScdException.class, () -> lnAdapter.checkExtRefInfoCoherence(extRefInfo));
 
         bindingInfo.setServiceType(null);
         bindingInfo.setIedName("IED_NAME1");
         bindingInfo.setLdInst("LD_INST12");
         bindingInfo.setLnInst("1");
         bindingInfo.setLnClass("ANCR");
         bindingInfo.setPrefix("PR");
-        extRef = assertDoesNotThrow(() ->lnAdapter.checkExtRefInfoCoherence(extRefInfo));
-        assertEquals(extRefInfo.getBindingInfo().getIedName(),extRef.getIedName());
+        extRef = assertDoesNotThrow(() -> lnAdapter.checkExtRefInfoCoherence(extRefInfo));
+        assertEquals(extRefInfo.getBindingInfo().getIedName(), extRef.getIedName());
 
         ExtRefSourceInfo sourceInfo = new ExtRefSourceInfo();
         sourceInfo.setSrcCBName("UNKNOWN_CB");
         extRefInfo.setSourceInfo(sourceInfo);
-        assertThrows(ScdException.class, () ->lnAdapter.checkExtRefInfoCoherence(extRefInfo));
+        assertThrows(ScdException.class, () -> lnAdapter.checkExtRefInfoCoherence(extRefInfo));
 
         sourceInfo.setSrcCBName("rpt1");
-        assertDoesNotThrow(() ->lnAdapter.checkExtRefInfoCoherence(extRefInfo));
+        assertDoesNotThrow(() -> lnAdapter.checkExtRefInfoCoherence(extRefInfo));
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideIncompleteExtRefInfo")
+    void should_throw_exception_when_trying_update_extRefSource_with_wrong_arguments(ExtRefInfo incompleteExtrefInfo) throws Exception {
+
+        //Given
+        SCL scd = SclTestMarshaller.getSCLFromFile("/scd-extref-cb/scd_get_cbs_test.xml");
+        SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
+
+        IEDAdapter iAdapter = assertDoesNotThrow(() -> sclRootAdapter.getIEDAdapterByName("IED_NAME1"));
+        LDeviceAdapter lDeviceAdapter = assertDoesNotThrow(() -> iAdapter.getLDeviceAdapterByLdInst("LD_INST11").get());
+        AbstractLNAdapter<?> lnAdapter = AbstractLNAdapter.builder()
+                .withLDeviceAdapter(lDeviceAdapter)
+                .withLnClass(TLLN0Enum.LLN_0.value())
+                .build();
+
+        //When
+        Assertions.assertThatThrownBy(() -> lnAdapter.updateExtRefSource(incompleteExtrefInfo))
+
+                //Then
+                .isInstanceOf(IllegalArgumentException.class);
+
```

Comment:
To test exceptions, I usually put a single comment 
```
//When & Then
```

 because it is a single statement.